### PR TITLE
clients/js: added regression tests for ethers v5 & ethers v6

### PR DESCRIPTION
**Pull request by CedarMist**
Closes: #169

 * Bumps package to v.1.3.1
 * removes web3.js from supported list (it can be added in future if required, but needs tests similar to ethers)
 * Updates CHANGELOG in clients/js
 * signed queries with Ethers v6.
 * fixed compat.ts tests for signers
 * removed peer dependencies for clients/js
 * added eqeqeq linter

----

While testing I encountered another problem: custom reverts are broken with Ethers v5. This is due to type confusion in Ethers v5 by then trying to work-around people avoiding type confusion in JSON-RPC endpoints.
See: https://github.com/ethers-io/ethers.js/blob/ec1b9583039a14a0e0fa15d0a2a6082a2f41cf5b/packages/providers/src.ts/json-rpc-provider.ts#L61

Custom reverts never worked with Ethers v5, but worked when using Hardhat with Ethers v5... Now we have tests for ethers v5 & ethers 6.

Note that reversion data returned by oasis-web3-gateway isn't encrypted. These errors are successfully caught by Ethers v6 and are not attempted to be decrypted by the sapphire wrapper. So the recommendation is to upgrade to Ethers v6.

### CODE REVIEW

File: examples/ethersv5-ts-esm/src/index.ts [hunk 1]
```diff
@@ -1,39 +1,83 @@
...
-import { TestErc20Token } from "./TestErc20Token.js";
 import { assert } from "console";
 
+import OmnibusJSON from "../../../contracts/artifacts/contracts/tests/Omnibus.sol/Omnibus.json" assert { type: "json" };
+
+async function testTheContract(contract:Contract, signerAddr?:string)
+{
+    const addr = await contract.callStatic["testSignedQueries()"]!();
+    if( signerAddr ) {
+        assert( addr == signerAddr );
```
Comment (lukaw3d) [inline examples/ethersv5-ts-esm/src/index.ts:13]: You might want https://eslint.org/docs/latest/rules/eqeqeq

Comment (CedarMist) [reply examples/ethersv5-ts-esm/src/index.ts]: Currently reviewing these

```
/src/sapphire-paratime/clients/js/lib/esm/cipher.js
  124:48  error  This generator function does not have 'yield'  require-yield
  129:48  error  This generator function does not have 'yield'  require-yield
  134:48  error  This generator function does not have 'yield'  require-yield
  167:48  error  This generator function does not have 'yield'  require-yield
  174:48  error  This generator function does not have 'yield'  require-yield
  187:48  error  This generator function does not have 'yield'  require-yield
  192:48  error  This generator function does not have 'yield'  require-yield
  217:65  error  This generator function does not have 'yield'  require-yield

/src/sapphire-paratime/clients/js/lib/esm/compat.js
   14:11   warning  Expected '!==' and instead saw '!='  eqeqeq
   15:25   error    'p' is already defined               no-redeclare
  300:70   warning  Expected '!==' and instead saw '!='  eqeqeq
  349:16   warning  Expected '!==' and instead saw '!='  eqeqeq
  413:122  error    'globalThis' is not defined          no-undef
  413:145  error    'globalThis' is not defined          no-undef
  413:178  error    'globalThis' is not defined          no-undef
  421:18   warning  Expected '===' and instead saw '=='  eqeqeq

/src/sapphire-paratime/clients/js/scripts/proxy.ts
   45:18  warning  Expected '!==' and instead saw '!='  eqeqeq
   71:23  warning  Expected '===' and instead saw '=='  eqeqeq
   74:21  warning  Expected '===' and instead saw '=='  eqeqeq
   75:21  warning  Expected '===' and instead saw '=='  eqeqeq
   85:34  warning  Expected '===' and instead saw '=='  eqeqeq
   88:29  warning  Expected '===' and instead saw '=='  eqeqeq
  112:30  warning  Expected '===' and instead saw '=='  eqeqeq
  117:27  warning  Expected '===' and instead saw '=='  eqeqeq

/src/sapphire-paratime/clients/js/src/compat.ts
  465:65  warning  Expected '!==' and instead saw '!='  eqeqeq
  529:14  warning  Expected '!==' and instead saw '!='  eqeqeq
  624:16  warning  Expected '===' and instead saw '=='  eqeqeq

/src/sapphire-paratime/clients/js/test/compat.spec.ts
  126:23  warning  Expected '===' and instead saw '=='  eqeqeq
  189:26  warning  Expected '===' and instead saw '=='  eqeqeq
```

File: examples/ethersv5-ts-esm/src/index.ts [hunk 2]
```diff
@@ -1,39 +1,83 @@
...
+        assert(false);
+    }
+    catch(e:any) {
+        console.log(e);
+        assert(e.code == 'CALL_EXCEPTION');
+        assert(e.errorName == 'CustomError');
+        assert((e.errorArgs[0] as BigNumber).toHexString()  == '0x1023456789abcdef1023456789abcdef1023456789abcdef1023456789abcdef');
+    }
+    */
+}
```
Comment (lubej) [inline examples/ethersv5-ts-esm/src/index.ts:34]: Should we keep this dead code in?

**Review by lubej — APPROVED** [review]

File: clients/js/.eslintrc.cjs [hunk 3]
```diff
@@ -20,5 +20,6 @@ module.exports = {
   rules: {
     '@typescript-eslint/no-explicit-any': 'off',
     '@typescript-eslint/no-non-null-assertion': 'off',
+    eqeqeq: 1,
```
Comment (lubej) [inline clients/js/.eslintrc.cjs:23]: Better to follow convention, I don't see 1 listed as one of the options: https://eslint.org/docs/latest/rules/eqeqeq#options

Comment (CedarMist) [reply clients/js/.eslintrc.cjs:23]: I got the following error in the CLI initially:

```
Error: .eslintrc.cjs:
	Configuration for rule "eqeqeq" is invalid:
	Severity should be one of the following: 0 = off, 1 = warn, 2 = error.
```

So I changed it to `1` to warn but not error out.

File: examples/ethersv5-ts-esm/src/index.ts (hunk 2 above)
Comment (CedarMist) [reply examples/ethersv5-ts-esm/src/index.ts:34]: Yes please, it's a tombstone to keep note that this specific behavior is known to be broken in Ethers v5. Where the ethers v5 & v6 tests are carbon copies of each other.

**Pull request merged**